Jenkinsfile

```diff
@@ -1,153 +1,103 @@
-        expression { env.BRANCH_NAME == 'latest' }
-      }
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
-        }
-      }
-      steps {
-        sh 'ls' // quick debug step
-        sh 'make storybook'
-        sh 'rm published.txt || true'
-        withCredentials([string(credentialsId: 'npm_bbc-online_read_write', variable: 'NPM_TOKEN')]) {
-          sh 'make publish'
-        }
-        stash name: 'psammead-publishes', includes: 'published.txt', allowEmpty: true
-        sh 'rm published.txt || true'
-      }
-      post {
-        always {
-          script {
-            stageName = env.STAGE_NAME
+
+          if (env.BRANCH_NAME == 'latest') {
+            stage ('Deploy Storybook & Publish to NPM') {
+              parallel (
+                'Deploy Storybook': {
+                  sh 'npm run deploy-storybook'
+                },
+                'Publish to NPM': {
+                    withCredentials([string(credentialsId: 'npm_bbc-online_read_write', variable: 'NPM_TOKEN')]) {
+                      sh 'make publish'
+                    }
+                }
+              )
+            }
+
+            stage ('Talos') {
+              sh 'git fetch --all'
+              sh 'make talos'
+            }
           }
-        }
-      }
-    }
-    stage ('Talos') {
-      when {
-        expression { env.BRANCH_NAME == 'latest' }
-      }
-      agent {
-        docker {
-          image "${nodeImage}"
-          label nodeName
-          args '-u root -v /etc/pki:/certs'
-        }
-      }
-      steps {
-        sh 'make setup-git'
-        sh 'git fetch --all'
-        unstash 'psammead-publishes'
-        sh 'make talos'
-      }
-      post {
-        always {
-          script {
-            stageName = env.STAGE_NAME
+        } catch (Throwable e) {
+          echo "Pipeline Failed: ${e}"
+
+          // send slack notification if building branch: latest
+          if (env.BRANCH_NAME == 'latest') {
+            // catch error in stages and notify slack
+            notifySlack('danger', 'Failed', gitCommitAuthor, e, gitCommitHash, gitCommitMessage, slackChannel)
+          }
+
+          // throw caught error to ensure pipeline fails
+          throw e
+        } finally {
+          cleanUp()
+
+          // send slack notification if building branch: latest
+          if (env.BRANCH_NAME == 'latest') {
+            switch (currentBuild.result) {
+              case 'SUCCESS':
+                notifySlack('good', 'Success', gitCommitAuthor, 'Successfully Deployed', gitCommitHash, gitCommitMessage, slackChannel)
+                break
+              case 'UNSTABLE':
+                notifySlack('DANGER', 'Unstable', gitCommitAuthor, 'Pipeline in an unstable state', gitCommitHash, gitCommitMessage, slackChannel)
+                break
+              default:
+                echo "Pipeline has failed with an unknown build result"
```

Comment:
This feels like something that should be slacked? I can see that this shouldn't happen, but I'm somewhat concerned that if nothing gets slacked then no one will notice that the pipeline is in a really weird state.

Reply:
Isn't this missing aborted? Or is that caught in the try/catch?

Reply:
The reason this isn't slacked is because an unstable build is caught in the try catch but this finally block will always run so would cause a double slack notification.

Reply:
But `UNSTABLE` is in the above case statement?
(I think you mean `FAILURE` or whatever the code is in Jenkins, in which case:)
could we not 
```
case 'FAILURE':
  break
default:
  notifySlack('DANGER', 'Unknown', ....)
  break
```

Reply:
Discussed in a DM and changes pushed